### ISSUE TITLE
[Bug] DeepSeek V4 常驻解码元数据导致 prepare_decode 耗时显著增长

### ISSUE THREAD

**Issue by high-cloud** [hw-native-sys/pypto-serving#198]
### Diagnosis

**pypto-serving** - `0ae4fd` 将 DeepSeek V4 fused MTP K=1 的 decode metadata 改为 device-resident，并在 `prepare_decode` 热路径中通过逐 rank、逐 tensor 的同步 `copy_to` 更新。当前 serving 的 grouped KV cache 会频繁改变 physical block IDs，导致 metadata 持续 dirty、流水并行被破坏。

### Description

DeepSeek V4 标准 GBS32 对齐测试中，commit `0ae4fd` 引入明显性能回归：

1. fused MTP K=1 的 8 个 decode metadata tensor 从 Host-shared 改为 device-resident：
   - `block_table`
   - `hca_cmp_block_table`
   - `csa_cmp_block_table`
   - `idx_block_table`
   - `hca_compress_state_block_table`
   - `csa_compress_state_block_table`
   - `csa_inner_compress_state_block_table`
   - `block_counts`
2. dirty key 包含所有 cache group 的完整 physical block ID 列表。
3. 任意一个 cache group 变化，都会将该 rank 的全部 8 个 metadata tensor 判为 dirty。
4. DP8 下所有 rank 同时 dirty 时，一次 prepare 会执行 `8 ranks × 8 tensors = 64` 次同步 `worker.copy_to()`。
5. 上传前还会执行 `predecessor.wait()`，等待上一轮 decode dispatch 完成，因此原本应与 dispatch 重叠的 `prepare_decode` 被串行化。
6. 两个 ping-pong slot 各自维护 resident metadata。CSA state 每约 4 token 变化一次，MTP 每步通常推进约 2 token，使两个 slot 几乎交替持续进入 dirty path。

观察到的组合测试结果：

| 配置 | prepare | dispatch | reclaim | Batch elapsed | 吞吐 |
|---|---:|---:|---:|---:|---:|
| 当前 serving + 当前 PyPTO | 176.9 ms | 76.2 ms | 48.9 ms | 29.77 s | 275.2 tok/s |
| 回退 serving + 当前 PyPTO | 14.2 ms | 57.8 ms | 79.2 ms | 14.90 s | 549.9 tok/s |
| 当前 serving + PyPTO b10 | 65.8 ms | 7.6 ms | 49.0 ms | 12.19 s | 672.3 tok/s |
| 回退 serving + PyPTO b10 | 15.6 ms | 8.0 ms | 53.0 ms | 9.55 s | 857.8 tok/s |

其中组合交互导致的额外 prepare 开销约为：

```text
176.9 - 65.8 - 14.2 + 15.6 = 112.5 ms
```

预期行为：

- `prepare_decode` 不应等待上一轮 device dispatch。
- 动态 block metadata 不应通过大量同步 control-plane `copy_to` 更新。
- 保留 fused K=1 的 write-only output/state resident 优化：
  - main `hidden_out`、`logits`
  - MTP `input_ids`、`position_ids`、`hidden_out`、`next_pre_hc_hidden`、`logits`
- 将上述 8 个动态 decode metadata tensor 恢复为 Host-shared。

### Command or Request

```bash
source /data/yangyaodong/pypto-serving-st-env-b10cae3d/activate.sh

bash .agents/skills/profile-dsv4-aligned-gbs32-strace/scripts/run_profile.sh \
  --model-dir /data/models/dsv4-flash-w8a8 \
  --devices 1,2,3,4,5,6,7,8 \
  --use-compile-cache \
  --artifact-dir /path/to/artifacts \
  --run-id resident-metadata-host-rollback
```

固定 workload：

```text
GBS32 / DP8 / EP8 / per-rank batch4
64 prompt tokens / 256 output tokens
MTP k=1 / temperature=0
chunked prefill enabled / prefix cache disabled
```

### Environment

| Component | Version |
|---|---|
| pypto-serving | `013cabe`，回归由 `0ae4fd` 引入 |
| pypto-lib | `2cd5f82` |
| pypto | `b10cae3d20a1` / package `0.1.0` |
| simpler | package `0.1.0` |
| ptoas | `0.57` |
| CANN | `9.0.0` |
| torch / torch-npu | `2.6.0+cpu` / `2.6.0.post5` |

### Host Platform

Linux (aarch64)

### Device / Platform

Ascend a2a3，DP8/EP8。

### Model

DeepSeek V4 Flash W8A8：

```text
/data/models/dsv4-flash-w8a8
```

### Logs

```text
当前 serving + 当前 PyPTO：
prepare_decode 约 176.9 ms
Batch elapsed 29.77 s
吞吐 275.2 tok/s

当前 serving + PyPTO b10：
prepare_decode 约 65.8 ms
Batch elapsed 12.19 s
吞吐 672.3 tok/s

回退 serving + PyPTO b10：
prepare_decode 约 15.6 ms
Batch elapsed 9.55 s
吞吐 857.8 tok/s
```

### Additional Context

`0ae4fd` 的 write-only output resident 优化方向正确，问题集中在动态 metadata resident 化以及同步 dirty 更新策略。

metadata 总传输量并不大，DP8 每次完整更新约 1.38 MB；主要开销来自：

- 64 次独立同步 `copy_to`
- `predecessor.wait()`
- ping-pong slot 的重复更新
- prepare/dispatch pipeline overlap 被破坏

建议先将全部 decode metadata 恢复为 Host-shared。未来若重新实现 resident metadata，应至少具备：

- per-tensor dirty tracking
- 合并或批量上传
- 不在 prepare 热路径等待 predecessor
- 两个 ping-pong slot 之间避免重复上传相同 metadata generation
